### PR TITLE
Update chunk_overlap in service_context.py

### PR DESCRIPTION
**Pull request by Miroun**
Chunk_overlap should be able to be set to 0.

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] I stared at the code and made sure it makes sense

### CODE REVIEW

File: llama_index/service_context.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def from_defaults(
             or node_parser
             or _get_default_node_parser(
                 chunk_size=chunk_size or DEFAULT_CHUNK_SIZE,
-                chunk_overlap=chunk_overlap or SENTENCE_CHUNK_OVERLAP,
+                chunk_overlap=SENTENCE_CHUNK_OVERLAP if chunk_overlap is None else chunk_overlap,
```
Comment (hatianzhang) [inline llama_index/service_context.py:208]: @logan-markewich since we are deprecating service context. better to let you handle it

Comment (hatianzhang) [reply llama_index/service_context.py]: @Miroun are those `chunk_overlap=chunk_overlap or SENTENCE_CHUNK_OVERLAP` the same as `chunk_overlap=SENTENCE_CHUNK_OVERLAP if chunk_overlap is None else chunk_overlap,
`

**Comment by logan-markewich** [comment]
Hmmm, it says this PR has no changes now?

**Comment by hatianzhang** [comment]
@logan-markewich yeah I fixed the lint errors and it turns out no change. not sure there is any different between original vs @Miroun 's change

**Pull request closed without merging**